src/pages/AddTransactionPage.ts

```diff
@@ -0,0 +1,114 @@
+import IPage from './IPage';
+import MenuPage from './MenuPage';
+import { IPrompter } from '../Prompter';
+import { IUserIO } from '../CLI';
+import { Answers } from 'inquirer';
+import { userTable, transactionTable } from '../tables';
+import { IUser } from '../tables/UserTable';
+import chalk from 'chalk';
+
+export default class AddTransactionPage implements IPage {
+  userIO: IUserIO;
+  prompter: IPrompter;
+  user: IUser;
+
+  constructor(userIO: IUserIO, prompter: IPrompter, user: IUser) {
+    this.userIO = userIO;
+    this.prompter = prompter;
+    this.user = user;
+  }
+
+  async display(): Promise<IPage> {
+    this.clearAndDisplayTitle();
+    const otherUser: IUser | undefined = await this.getUser();
+
+    if (!otherUser) {
+      throw Error('The other user cannot be undefined.')
+    }
+
+    const name: string = await this.getName();
+    const date: Date = await this.getDate();
+    const userPaid: boolean = await this.getUserPaid();
+    const cost: number = await this.getCost();
+
+    const lender: IUser = userPaid ? this.user : otherUser;
+    const borrower: IUser = userPaid ? otherUser : this.user;
```

Comment:
is it possible to check `userPaid` just once? might it be the case to use destructuring assignment? (I'm not sure, just wondering)